mobile/src/navigations/root/root.tsx

```diff
@@ -22,41 +20,47 @@ const screenOptions: NativeStackNavigationOptions = {
 
 const Root: React.FC = () => {
   const dispatch = useAppDispatch();
-  const { isSurveyCompleted, authenticatedUser, authenticatedUserDataStatus } =
-    useAppSelector(({ auth }) => {
-      return {
-        isSurveyCompleted: auth.authenticatedUser?.isSurveyCompleted,
-        authenticatedUser: auth.authenticatedUser,
-        authenticatedUserDataStatus: auth.authenticatedUserDataStatus,
-      };
-    });
+  const {
+    authenticatedUser,
+    authenticatedUserDataStatus,
+    surveyPreferencesDataStatus,
+  } = useAppSelector(({ auth }) => {
+    return {
+      authenticatedUser: auth.authenticatedUser,
+      authenticatedUserDataStatus: auth.authenticatedUserDataStatus,
+      surveyPreferencesDataStatus: auth.surveyPreferencesDataStatus,
+    };
+  });
+
+  const isLoaderVisible =
+    authenticatedUserDataStatus === DataStatus.IDLE ||
+    authenticatedUserDataStatus === DataStatus.PENDING ||
+    surveyPreferencesDataStatus === DataStatus.PENDING;
 
   useEffect(() => {
     void dispatch(authActions.getAuthenticatedUser());
   }, [dispatch]);
 
-  if (
-    authenticatedUserDataStatus === DataStatus.IDLE ||
-    authenticatedUserDataStatus === DataStatus.PENDING
-  ) {
-    return <Loader />;
-  }
-
   return (
-    <NativeStack.Navigator screenOptions={screenOptions}>
-      {authenticatedUser ? (
-        isSurveyCompleted ? (
-          <NativeStack.Screen name={RootScreenName.MAIN} component={Main} />
-        ) : (
-          <NativeStack.Screen name={RootScreenName.SURVEY} component={Survey} />
-        )
-      ) : (
-        <>
-          <NativeStack.Screen name={RootScreenName.SIGN_IN} component={Auth} />
-          <NativeStack.Screen name={RootScreenName.SIGN_UP} component={Auth} />
-        </>
-      )}
-    </NativeStack.Navigator>
+    <>
+      <NativeStack.Navigator screenOptions={screenOptions}>
+        {NAVIGATION_ITEMS.filter((screen) => {
+          return screen.conditionToRender(
+            Boolean(authenticatedUser),
+            Boolean(authenticatedUser?.isSurveyCompleted),
+          );
+        }).map((screen) => {
```

Comment:
Please put filtering part into variable above, separately from render part and wrap with useMemo